xgboost_ray/data_sources/partitioned.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any, Optional, Sequence, Dict, Union, Tuple
+
+from collections import defaultdict
+import pandas as pd
+import numpy as np
+
+import ray
+from ray import ObjectRef
+from ray.actor import ActorHandle
+
+from xgboost_ray.data_sources._distributed import \
+    assign_partitions_to_actors, get_actor_rank_ips
+from xgboost_ray.data_sources.data_source import DataSource, RayFileType
+from xgboost_ray.data_sources.object_store import ObjectStore
+from xgboost_ray.data_sources.pandas import Pandas
+from xgboost_ray.data_sources.numpy import Numpy
+
+
+class Partitioned(DataSource):
+    """Read from distributed data structure implementing __partitioned__.
+
+    __partitioned__ provides meta data about how the data is partitioned and
+    distributed across several compute nodes, making supporting objects them
+    suitable for distributed loading.
```

Comment:
Could we link to the spec here as in https://github.com/ray-project/xgboost_ray/issues/152?

Reply:
Yes, we could. I did not because we hope to find a better home for the spec (like in the pydata API universe or at least the main branch there) at some point. Let me add it here nevertheless, I will record implementations that we can scan for reference should we move the spec.

Reply:
Done.